### PR TITLE
 add actions in app

### PR DESCRIPTION
**Pull request by remoteretrieval**
## Summary by CodeRabbit

- **New Features**
	- Introduced the Remote-Retriever API for automated contact information retrieval, enhancing CRM data and lead generation tools.
	- Added functionality for retrieving pending orders and specific device return orders from a remote service.
	- Implemented utility functions for handling data streams and URL parameters.
	- Enhanced API interactions with new methods and property definitions for improved order retrieval.

- **Documentation**
	- Added comprehensive README documentation for the Remote-Retriever API, outlining its functionality and use cases.

- **Chores**
	- Centralized configuration values via a constants file for improved maintainability.

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import app from "../../remote_retriever.app.mjs";
+
+export default {
+  key: "remoteretriever-get-pending-order",
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs:4]:
```suggestion
  key: "remote_retrieval-get-pending-order",
```

File: components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name: "Get Pending Order",
+  description: "These are the orders for which the payment process has not been completed.[See the documentation](https://www.remoteretrieval.com/api-documentation/#pending-orders)",
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs:6]:
```suggestion
  description: "Retrieve a list of the orders for which the payment process has not been completed.[See the documentation](https://www.remoteretrieval.com/api-documentation/#pending-orders)",
```

File: components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  props: {
+    app,
+  },
+  methods: {
+    getPendingOrder() {
+      return this.app.makeRequest({
+        path: "/pending-orders/",
+      });
+    },
+  },
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs:18]: The method `pendingOrders` using the endpoint `/pending-orders/` already exists within the app file (remote_retrieval.app.mjs).
```suggestion
```

File: components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  async run({ $: step }) {
+    const { getPendingOrder } = this;
+    const response = await getPendingOrder({
+      step,
+    });
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs:23]:
```suggestion
    const response = await this.app.pendingOrders({
      step,
    });
```

File: components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    step.export("$summary", `Successfully retrieved order with ID \`${response.id}\`.`);
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-pending-order/get-pending-order.mjs:25]:
```suggestion
    step.export("$summary", `Successfully retrieved ${response.length} order(s).`);
```

File: components/remote_retrieval/actions/get-specific-order/get-specific-order.mjs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import app from "../../remote_retriever.app.mjs";
+
+export default {
+  key: "remote_retrieval-get-specificorder",
```
Comment (michelle0927) [inline components/remote_retrieval/actions/get-specific-order/get-specific-order.mjs:4]:
```suggestion
  key: "remote_retrieval-get-specific-order",
```

File: components/remote_retrieval/remote_retrieval.app.mjs [hunk 7]
```diff
@@ -1,11 +1,133 @@
...
+        }
+
+        const {
+          next,
+          results: orders,
+        } = await this.pendingOrders({
+          params: {
+            page,
+          },
+        });
```
Comment (michelle0927) [inline components/remote_retrieval/remote_retrieval.app.mjs:26]: Are you sure that the response contains `next` and `results`? I haven't tested it, but the [documentation](https://www.remoteretrieval.com/api-documentation/#pending-orders) shows that this endpoint just returns an array.

File: components/remote_retrieval/remote_retrieval.app.mjs [hunk 8]
```diff
@@ -1,11 +1,133 @@
...
+    async *getResourcesStream({
+      resourceFn,
+      resourceFnArgs,
+      resourceName,
+      max = constants.DEFAULT_MAX,
+    }) {
+      let page;
+      let resourcesCount = 0;
+
+      while (true) {
```
Comment (michelle0927) [inline components/remote_retrieval/remote_retrieval.app.mjs:97]: I don't see the method `getResourceStream()` being called anywhere. I assume it's meant to be used in `get-pending-order`?

**Review by michelle0927 — CHANGES_REQUESTED** [review]
Thank you so much for your contribution! I left some comments/questions. You'll also need to update the version in `package.json` from `0.0.1` to `0.1.0`.
Since the action "Get Pending Order" is actually retrieving a list of orders, I would suggest changing the name to "Get Pending Orders" and updating the folder/file names and component key as well.
Let me know if you have any questions or would like help with implementing these changes.

**Review by remoteretrieval — COMMENTED** [review]
updates after review

**Review by michelle0927 — APPROVED** [review]
Approved! Ready for QA!

**Comment by remoteretrieval** [comment]
When it will show in apps directory ? Please tell me next step.

**Comment by michelle0927** [comment]
@remoteretrieval The actions are now published, and will appear in the Pipedream UI. You can view them towards the bottom of this page: https://pipedream.com/apps/remote-retrieval
Thanks again for your contribution!

**Pull request merged**